tools/src/h5ls/h5ls.c

```diff
@@ -2995,23 +3002,27 @@ main(int argc, char *argv[])
     }
 
     /* Setup a custom fapl for file accesses */
-    if (custom_vol_fapl || custom_vfd_fapl) {
 #ifdef H5_HAVE_ROS3_VFD
-        if (custom_vfd_fapl && (0 == strcmp(vfd_info.u.name, drivernames[ROS3_VFD_IDX]))) {
-            if (!vfd_info.info)
-                vfd_info.info = &ros3_fa;
-        }
+    if (custom_vfd_fapl && (0 == strcmp(vfd_info.u.name, drivernames[ROS3_VFD_IDX]))) {
+        if (!vfd_info.info)
+            vfd_info.info = &ros3_fa;
+    }
 #endif
 #ifdef H5_HAVE_LIBHDFS
-        if (custom_vfd_fapl && (0 == strcmp(vfd_info.u.name, drivernames[HDFS_VFD_IDX]))) {
-            if (!vfd_info.info)
-                vfd_info.info = &hdfs_fa;
-        }
+    if (custom_vfd_fapl && (0 == strcmp(vfd_info.u.name, drivernames[HDFS_VFD_IDX]))) {
+        if (!vfd_info.info)
+            vfd_info.info = &hdfs_fa;
+    }
 #endif
 
-        if ((fapl_id = h5tools_get_fapl(H5P_DEFAULT, custom_vol_fapl ? &vol_info : NULL,
-                                        custom_vfd_fapl ? &vfd_info : NULL)) < 0) {
-            error_msg("failed to setup file access property list (fapl) for file\n");
+    if ((fapl_id = h5tools_get_fapl(H5P_DEFAULT, custom_vol_fapl ? &vol_info : NULL,
```

Comment:
Same comment here about this now always creating a fapl, which affects the logic used for `h5tools_fopen`.

Reply:
Yes, I found those concerning, too.